frontend/src/screens/studies/participant-controls.tsx

```diff
@@ -0,0 +1,195 @@
+    const onInputChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+        if (ev.target.checked) {
+            onChange({
+                ...state,
+                subjects: (
+                    state?.subjects ? state.subjects.concat(ev.target.value as any) : [ev.target.value]
+                ) as any as StudySubjectID[] })
+        } else {
+            onChange({ ...state, subjects: without<StudySubjectID>(state.subjects, ev.target.value as any) })
+        }
+    }
+    return (
+        <div css={{
+            input: {
+                marginRight: '0.5rem',
+            },
+        }}>
+            {Object.entries(StudySubjects).map(([value, label]) => (
+                <label key={value} className="dropdown-item">
+                    <input
+                        type="checkbox"
+                        value={value || ''}
+                        checked={state.subjects?.includes(value as any) || false}
+                        onChange={onInputChange}
+                    />
+                    {label}
+                </label>
+            ))}
+        </div>
+    )
+}
+
+export const DesktopFilterRow:React.FC<ControlProps> = ({ onChange, state, state: { subjects } } ) => {
+
+    if (!subjects?.length) return null
+
+    return (
+        <Box
+            css={{
+                borderBottom: '1px solid lightgray',
+                marginBottom: '1rem',
+            }}
+        >
+            <Box gap flex>
+                {subjects.map(subj => (
+                    <button key={subj}
+                        css={{ alignSelf: 'center' }}
+                        className="btn badge bg-light text-dark border border-1"
+                        onClick={() => onChange({ ...state, subjects: without<StudySubjectID>(subjects, subj) })}
+                    >
+                        {StudySubjects[subj]}
+                        <Icon icon="x" />
+                    </button>
+                ))}
+            </Box>
+            <Button onClick={() => onChange({ ...state, subjects: [] })}>
+                Clear Filters
+                <Icon icon="x" />
+            </Button>
+        </Box>
+    )
+}
+
+export const applyControls = ({ sort, subjects } : ControlState, studies: ParticipantStudy[]) => {
+    if (sort) {
+        if (sort == 'study:type') {
+            studies = sortBy(studies, (s) => tagOfType(s, 'type'))
+        } else if (sort == 'time:high-low') {
+            studies = sortBy(studies, s => s.durationMinutes * -1)
+        } else if (sort == 'time:low-high') {
+            studies = sortBy(studies, 'durationMinutes')
+        } else if (sort == 'points:high-low') {
+            studies = sortBy(studies, s => (s?.participationPoints || 0)* -1)
+        } else if (sort == 'points:low-high') {
+            studies = sortBy(studies, 'participationPoints')
+        }
+    }
+    if (subjects?.length) {
+        studies = studies.filter(s => subjects.find(subj => -1 != s.tags.indexOf(subj)))
```

Comment:
I'd write `find(subj => s.tags.includes(subj)` here, I think.

Reply:
good point, didn't think of that and reached for the old `indexOf` instead. I'll update, thank you!